upstairs/src/client.rs

```diff
@@ -2623,8 +2611,12 @@ pub(crate) fn validate_encrypted_read_response(
         // expect to see this case unless this is a blank block.
         //
         // XXX if it's not a blank block, we may be under attack?
-        assert!(response.data[..].iter().all(|&x| x == 0));
-        return Ok(None);
+        if response.data[..].iter().all(|&x| x == 0) {
+            return Ok(None);
+        } else {
+            error!(log, "got empty block context with non-blank block");
+            return Err(CrucibleError::DecryptionError);
```

Comment:
We should have a different error variant for this

Reply:
Sure, done in 264cbeff4edc63aa13b70721dd6fae65de66df88